### PR TITLE
Removed ngx-bootstrap from hostedTools

### PR DESCRIPTION
**Pull request by ByteMap**
dockstore/dockstore#3251

Migrated from ngx-bootstrap tabs to Material tabs

Removed ngx-bootstrap module from spec file

Changed Cypress tests to work for the new Material tabs for
hostedTools

### CODE REVIEW

**Comment by ByteMap** [comment]
How the new Material Tabs would look.
Screen Shot 2020-03-05 at 1 24 11 PM 1
Screen Shot 2020-03-05 at 1 24 31 PM
Screen Shot 2020-03-05 at 1 24 40 PM

**Comment by coverbeck** [comment]
Can you please attach a screenshot?

**Comment by ByteMap** [comment]
Also it seems that the Cypress test seems a little bit flaky especially at commands such as line 83. When getting the 'tool-path', the data is slow at updating itself. For instance, when executing the command `cy.get('#tool-path').contains('quay.io/hosted-tool/ht:1');`, ht does not update to ht1 until a few seconds later, which causes the test to fail. The only solution that I can think of is to extend `cy.wait()`, but that should be a last resort. Does anyone have any solution in mind besides extending `cy.wait()`?

**Comment by agduncan94** [comment]
@ByteMap Looks good but I would add some padding so that the dropdown for descriptor type isn't so far left.

**Comment by ByteMap** [comment]
Newest preview for the slight button and dropdown adjustments

For the tool-file-editor:
Screen Shot 2020-03-20 at 3 09 49 PM
Screen Shot 2020-03-20 at 3 10 01 PM

For the workflow-file-editor:
Screen Shot 2020-03-20 at 3 12 47 PM
Screen Shot 2020-03-20 at 3 12 56 PM

File: src/app/container/tool-file-editor/tool-file-editor.component.html [hunk 1]
```diff
@@ -14,54 +14,60 @@
...
+        [sourcefiles]="dockerFile"
+        [editing]="editing"
+        [fileType]="'dockerfile'"
+        [selectedVersion]="currentVersion"
+        [entryType]="'tool'"
+        [entrypath]="entrypath"
+      ></app-code-editor-list>
+    </ng-template>
+  </mat-tab>
+  <mat-tab label="Descriptor Files" id="descriptorFilesTab">
```
Comment (garyluu) [inline src/app/container/tool-file-editor/tool-file-editor.component.html]: unused id here and elsewhere

File: src/app/container/tool-file-editor/tool-file-editor.component.scss [hunk 2]
```diff
@@ -5,3 +5,7 @@
 .editor-button-container {
   margin-top: 10px;
 }
+
+.mat-form-field {
+  padding-left: 0.5rem;
```
Comment (garyluu) [inline src/app/container/tool-file-editor/tool-file-editor.component.scss]: this is just pl-2

File: src/app/shared/code-editor-list/code-editor-list.component.scss [hunk 3]
```diff
@@ -5,3 +5,7 @@
 .link-button {
   color: inherit;
 }
+
+.add-file-button {
+  left: 0.5rem;
```
Comment (garyluu) [inline src/app/shared/code-editor-list/code-editor-list.component.scss]: just use pl-2 if possible

Comment (ByteMap) [reply src/app/shared/code-editor-list/code-editor-list.component.scss]: The button is not affected by pl-2, so instead I have used ml-2.

File: src/app/container/tool-file-editor/tool-file-editor.component.scss (hunk 2 above)
Comment (ByteMap) [reply src/app/container/tool-file-editor/tool-file-editor.component.scss]: Done

File: src/app/container/tool-file-editor/tool-file-editor.component.html (hunk 1 above)
Comment (ByteMap) [reply src/app/container/tool-file-editor/tool-file-editor.component.html]: Removed the unused IDs

**Pull request merged**